collective/ogtags/browser/viewlets.py

```diff
@@ -79,24 +79,28 @@ def image_tags(self):
             return
         tags = []
         context = aq_inner(self.context)
+        field = None
         try:
             scales = context.restrictedTraverse('@@images', None)
         except (AttributeError, KeyError):
             scales = None
         if not scales:
             return self.default_image(self.settings.default_img)
         try:
-            image = context.image
-            field = 'image'
-        except AttributeError:
+            image_provider = IOGTagsImageProvider(self.context)
+            image = image_provider.getOGTagsImage()
```

Comment:
You will need to call `field = image.getName()` or something else to get the name of the field (working for Archetypes and dexterity). Or maybe let the provider return both an image and a fieldname.

Otherwise, when the code reaches the line `fieldname = field or 'image'` a bit further down, fieldname will always be `image`, because `field` is `None`.